src/cargo/util/toml/mod.rs

```diff
@@ -105,61 +105,62 @@ fn read_manifest_from_str(
 
     let mut unused = BTreeSet::new();
     let deserializer = toml::de::Deserializer::new(contents);
-    let manifest: manifest::TomlManifest = match serde_ignored::deserialize(deserializer, |path| {
-        let mut key = String::new();
-        stringify(&mut key, &path);
-        unused.insert(key);
-    }) {
-        Ok(manifest) => manifest,
-        Err(e) => {
-            let Some(span) = e.span() else {
-                return Err(e.into());
-            };
+    let mut manifest: manifest::TomlManifest =
```

Comment:
nit: ... so all of this was changed just from adding `mut`?  Ouch.

Might be worth a `let mut manifest = manifest` or a `let manifest = {... do changes; manifest};` to reduce churn

Reply:
> I had migrated the code to here, because detailed_dep_to_dependency will invoke by read_manifest_from_str.
These code had deleted.